hack/testing/logging.sh

```diff
@@ -0,0 +1,239 @@
+                "running"|"complete"|"pending")
+                    echo build in progress for $bc - delete failed build $build
+                    # delete the failed build - otherwise it will show up in the list and
+                    # the main loop will never Complete
+                    oc delete build $build
+                    needbuild=0
+                    break
+                    ;;
+                "failed")
+                    # if the build failed, there will be at least 1 failed status
+                    # if there is another build running or complete, it will be
+                    # detected above
+                    needbuild=1
+                    ;;
+                esac
+            done
+            # if we are here and needbuild=1, there were no running or complete builds
+            if [ $needbuild = "1" ] ; then
+                # start a new build
+                oc start-build $bc
+            fi
+        done
+        sleep $interval
+        waittime=`expr $waittime - $interval`
+    done
+    if [ $complete = 0 ] ; then
+        echo error builds are not complete
+        oc get builds
+        return 1
+    fi
+    return 0
+}
+
+trap "exit" INT TERM
+trap "cleanup" EXIT
+
+echo "[INFO] Starting logging tests at " `date`
+
+ensure_iptables_or_die
+os::util::environment::setup_all_server_vars "test-extended/logging/"
+os::util::environment::use_sudo
+reset_tmp_dir
+
+os::log::start_system_logger
+
+configure_os_server
+start_os_server
+
+export KUBECONFIG="${ADMIN_KUBECONFIG}"
+
+install_registry
+wait_for_registry
+
+######### logging specific code starts here ####################
+
+### create and deploy the logging component pods ###
+masterurlhack=",MASTER_URL=https://172.30.0.1:443"
+OS_O_A_L_DIR=${OS_O_A_L_DIR:-$OS_ROOT/test/extended/origin-aggregated-logging}
+os::cmd::expect_success "oc new-project logging"
+os::cmd::expect_success "oc secrets new logging-deployer nothing=/dev/null"
+os::cmd::expect_success "echo 'apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: logging-deployer
+secrets:
+- name: logging-deployer
+' | oc create -f -"
+os::cmd::expect_success "oc policy add-role-to-user edit system:serviceaccount:logging:logging-deployer"
+if [ -n "$USE_LOGGING_DEPLOYER" ] ; then
+    imageprefix="docker.io/openshift/origin-"
+elif [ -n "$USE_LOGGING_DEPLOYER_SCRIPT" ] ; then
+    pushd $OS_O_A_L_DIR/deployment
+    IMAGE_PREFIX="openshift/origin-" PROJECT=logging ./run.sh
+    popd
+    imageprefix=
+else
+    os::cmd::expect_success "oc process \
+       -f $OS_O_A_L_DIR/hack/templates/dev-builds.yaml \
+       -v LOGGING_FORK_URL=$GIT_URL,LOGGING_FORK_BRANCH=$GIT_BRANCH \
+       | oc create -f -"
+    os::cmd::expect_success "wait_for_builds_complete"
+    imageprefix=`oc get is | awk '$1 == "logging-deployment" {print gensub(/^([^/]*\/logging\/).*$/, "\\\1", 1, $2)}'`
+    sleep 5
+fi
+log_priv_user=$(create_valid_file add-privileged-logging-user.yml)
```

Comment:
Where what file is?  add-privileged-logging-user.yml is a temporary file created for use with the following code.